collections-csharp/MergingDictionariesInCSharp/MergingDictionariesInCSharp.Benchmark/MergingMethodsBenchmark.cs

```diff
@@ -11,45 +12,29 @@ public class MergingMethodsBenchmark
     private Dictionary<int, string> _dictionaryB;
 
     [GlobalSetup]
-    private void Setup()
+    public void Setup()
     {
         _dictionaryA = new Dictionary<int, string>();
         _dictionaryB = new Dictionary<int, string>();
 
-        Random random = new Random();
-
         // Generate random values for dictionaryA
-        for (int i = 0; i < 100; i++)
+        for (var i = 0; i < 100; i++)
         {
-            int key = random.Next(100);
-            string value = GenerateRandomValue();
-
-            if (!_dictionaryA.ContainsKey(key))
-            {
-                _dictionaryA.Add(key, value);
-            }
-            else
-            {
-                i--; // Retry generating a unique key
-            }
+            var randomValue = GenerateRandomValue();
+            _dictionaryA.Add(i, randomValue);            
         }
 
         // Generate random values for dictionaryB
-        foreach (var kvp in _dictionaryA)
+        for (int i = 100; i < 200; i++)
         {
-            int key = random.Next(100, 200);
-            string value = GenerateRandomValue();
-
-            if (!_dictionaryB.ContainsKey(key) && !_dictionaryA.ContainsKey(key))
-            {
-                _dictionaryB.Add(key, value);
-            }
+            var randomValue = GenerateRandomValue();
+            _dictionaryB.Add(i, randomValue);            
         }
     }
 
     private string GenerateRandomValue()
     {
-        Random random = new Random();
+        var random = Random.Shared;
```

Comment:
No need for the variable, just use Random.Shared directly. It is static and thread safe.